001-finding_resources.md

```diff
@@ -0,0 +1,120 @@
+# Design Document 001 - Finding Resources
+
+TODO: Overview
+
+## Finding SDF URI Resources
+
+The following sections discuss how `<uri>` elements in SDF files are
+resovled in order to find meshes and models. This section does not cover
+finding other resources, such as plugins.
+
+### Current Process
+
+This section describes how Ignition Gazebo currently finds Mesh URI
+resources. SDF contains multiple instances of URI elements, but the current
+implementation of Gazebo handles only mesh URIs. The information contained
+here can be extrapolated to handle other types of resources, such as
+textures and audio files.
+
+A user of Ignition may not need or use Gazebo. If this describes you, then
+skip directly to the last step which describes how
+`ignition::common::findFile` can be used to resolve a resource.
+
+1. Ignition Gazebo registers an SDF find callback in the Server constructor.
+   This callback is triggered by SDF whenever a URI resource from an
+   `<include>` element cannot be found. The Ignition Gazebo find callback in
```

Comment:
Not sure if you want to also mention other ways in which SDF may find the file before it triggers the callback. They're all documented [here](http://sdformat.org/tutorials?tut=composition&cat=specification&branch=model_comp_legacy#including-a-model).

Reply:
Add in a3696f8e717265fb69c906bc630f83723b50d8dd